src/components/no-match.js

```diff
@@ -14,8 +16,13 @@ import {
 import main from '../css/app.css'
 import blocks from '../css/no-match.css'
 
-export default class NoMatch extends React.PureComponent {
+class NoMatch extends React.PureComponent {
+  componentDidMount() {
+    this.props.whoami()
```

Comment:
Did you test this PR?

I suggested that if we can use the already fetched `whoAmI` response, calling an API for every 404 pages is not a good idea in my opinion, especially when we have called it already and have stored it with us.